core/src/Kokkos_Parallel.hpp

```diff
@@ -137,9 +137,9 @@ inline void parallel_for(const std::string& str, const ExecPolicy& policy,
   ExecPolicy inner_policy = policy;
   Kokkos::Tools::Impl::begin_parallel_for(inner_policy, functor, str, kpID);
 
-  Kokkos::Impl::shared_allocation_tracking_disable();
-  Impl::ParallelFor<FunctorType, ExecPolicy> closure(functor, inner_policy);
-  Kokkos::Impl::shared_allocation_tracking_enable();
+  auto closure = Kokkos::Impl::with_shared_allocation_tracking_disabled([&]() {
+    return Impl::ParallelFor<FunctorType, ExecPolicy>(functor, inner_policy);
+  });
```

Comment:
What is the rational behind this change? I don't necessarily disagree with it but not clear how this relates to mdspan?

Reply:
That's https://github.com/kokkos/kokkos/pull/6289.

Reply:
Yeah my branch is based on top of that please ignore and prefer #6289